### DISCUSSION TITLE
Switches to a wrong window after a password entry popup

### DISCUSSION THREAD

**Discussion by nickkadutskyi** [nikitabobko/AeroSpace#952] [potential-bugs]
**Describe the bug**
I have this integration in my shell where sudo prompts FaceId entry on my Mac, also I have 1Password ssh integration which also creates a popup for approval when I run ssh commands.
The problem is that when I type a password or use fingerprint reader popup is closing and I focus switches to a wrong terminal window.
Here is a screencast:
https://github.com/user-attachments/assets/8e6df2d5-9d90-48f6-a798-4485c5d2e692
Here is another screencast with more windows:
https://github.com/user-attachments/assets/36614853-9712-4612-be9d-31a9c5f00c71
It seems like it switches to the last opened window in that case and not the last focused.

**Related issues you could find**
This issue might be related: https://github.com/nikitabobko/AeroSpace/issues/571

**Comment by ntawileh** [comment]
I have a possibly related issue also involving the Mac password popup.  I typically have different Chrome windows in different spaces (work vs. personal).  

Chrome activity triggers the Mac sudo popup (like Chrome needing to pull a password for a site I'm logging in to)
I auth using the Mac sudo popup.  As soon as that popup closes, aerospace pulls a Chrome window that was in another space and brings it in to the current space.  Either that, or the current chrome window that I was in just moves to another space where other Chrome windows are.

**Comment by obsidian33** [comment]
I think my issue may be related. To this just a slightly different scenario.

When I have an dialog that pops up, for example when moving an app to the trash, it prompts for the user and password. I have this account in 1password so I typically use autofill for this. When I trigger 1password autofill another 1password dialog appears and when I select a credential record the focus is returned from the 1password dialog back to another window and NOT the apple dialog prompting for the username and password. This effectively prevents the credentials from autofilling in to the correct window and instead can be entered in to separate application.
